docs/proposals/transformer-recursion-depth-budget.md

```
@@ -0,0 +1,229 @@
+    if isinstance(template, dict):
+        if self.marker in template: return self.walk_rule(template, context, path)
+        return self.walk_dict(template, context, path)
+    return self.walk_scalar(template, context, path)
+```
+
+So **every** descent through a template node costs a `walk` **and** a `_walk` frame where one would
+do. This is not marginal, because the codec walks a *rule at every level*. Profiling the committed
+editor encoder over a 30-deep input (peak 801 live frames) shows the recursion is dominated by a
+lockstep triple:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)